stubs/jsonschema/jsonschema/exceptions.pyi

```diff
@@ -1,40 +1,50 @@
+from _typeshed import Self
+from collections import deque
+from collections.abc import Callable, Container, Iterable, Sequence
 from typing import Any
 
-WEAK_MATCHES: Any
-STRONG_MATCHES: Any
+from jsonschema import _utils, protocols
+
+_RelevanceFuncType = Callable[[ValidationError], Any]
```

Comment:
Functions [passed to the `key` argument of `builtins.max`](https://github.com/python-jsonschema/jsonschema/blob/65f591e5152e37867cdd5db1baf6ae175229713e/jsonschema/exceptions.py#L359) must return objects that satisfy the `_typeshed.SupportsRichComparison` interface:

https://github.com/python/typeshed/blob/2d2b34c1ee25f9c63554250d2b5f897a395ae300/stdlib/builtins.pyi#L1295

```suggestion
_RelevanceFuncType = Callable[[ValidationError], SupportsRichComparison]
```

(You'll have to import it from `_typeshed` at the top of the file.)

Reply:
Ah, that's why `Ctrl+F: "Comparable"` didn't get me anything on the typing docs!

I wanted this but decided against adding
```python
class _Comparable(Protocol):
    def __lt__(self, other: Any) -> bool: ...
```
to the stubs.

I'll make the change, with great enthusiasm.

(Is `SupportsRichComparison` on a path towards inclusion in `typing-extensions` and `typing`?)

Reply:
> Is `SupportsRichComparison` on a path towards inclusion in `typing-extensions` and `typing`?

I'm rooting for it 🤞 but I think it's still too new to add it to `typing_extensions` just yet — we only added it a few months ago in #6583, so I don't think it can really be considered stable yet